File: domain-driven-design/js/domain/order.js
// 领域层，包含核心的业务模型和逻辑
// domain/order.js
export class Order {
  constructor(id, customerName, amount) {
    this.id = id;
    this.customerName = customerName;
    this.amount = amount;
    this.status = 'CREATED';
  }

  cancel() {
    if (this.status === 'CREATED') {
      this.status = 'CANCELED';
      console.log(`订单 ID ${this.id} 已取消`);
    } else {
      console.log(`订单 ID ${this.id} 已经取消，无法重复操作`);
    }
  }
}

File: domain-driven-design/java/domain/Order.java
// domain/Order.java
package domain;

public class Order {
    private int id;
    private String customerName;
    private double amount;
    private String status;

    public Order(int id, String customerName, double amount) {
        this.id = id;
        this.customerName = customerName;
        this.amount = amount;
        this.status = "CREATED";
    }

    public void cancel() {
        if ("CREATED".equals(this.status)) {
            this.status = "CANCELED";
            System.out.println("订单 ID " + this.id + " 已取消");
        } else {
            System.out.println("订单 ID " + this.id + " 已经取消，无法重复操作");
        }
    }

    public void display() {
        System.out.println("订单 ID: " + this.id);
        System.out.println("客户名称: " + this.customerName);
        System.out.println("订单金额: " + String.format("%.2f", this.amount));
        System.out.println("订单状态: " + this.status);
    }

    public int getId() {
        return id;
    }
}